test/visual/renderer.hpp

```diff
@@ -191,19 +203,88 @@ struct grid_renderer : renderer_base<mapnik::image_rgba8>
 };
 #endif
 
+template <typename T>
+void set_rectangle(T const & src, T & dst, std::size_t x, std::size_t y)
+{
+    mapnik::box2d<int> ext0(0, 0, dst.width(), dst.height());
+    mapnik::box2d<int> ext1(x, y, x + src.width(), y + src.height());
+
+    if (ext0.intersects(ext1))
+    {
+        mapnik::box2d<int> box = ext0.intersect(ext1);
+        for (std::size_t pix_y = box.miny(); pix_y < box.maxy(); ++pix_y)
```

Comment:
Careful with this comparison: `pix_y` is unsigned while `miny` is signed. This will soon be a warning we have on by default: #2907